src/mgmt/srv_drpc.c

```diff
@@ -252,43 +195,31 @@ ds_mgmt_drpc_create_mgmt_svc(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 void
 ds_mgmt_drpc_start_mgmt_svc(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 {
-	Mgmt__DaosResp	*resp = NULL;
-	int rc;
+	Mgmt__DaosResp	resp = MGMT__DAOS_RESP__INIT;
+	int		rc;
 
 	D_INFO("Received request to start MS\n");
 
-	D_ALLOC_PTR(resp);
-	if (resp == NULL) {
-		drpc_resp->status = DRPC__STATUS__FAILURE;
-		D_ERROR("Failed to allocate daos response ref\n");
-		return;
-	}
-
-	/* Response status is populated with SUCCESS on init. */
-	mgmt__daos_resp__init(resp);
-
 	rc = ds_mgmt_svc_start(false /* !create */, 0 /* size */,
 			       false /* !bootstrap */, NULL /* uuid */,
 			       NULL /* addr */);
-	if (rc == -DER_ALREADY) {
+	if (rc == -DER_ALREADY)
```

Comment:
I think in this case we should set rc = 0, since there was nothing to do. The old logic only set rc in the other error case.

Reply:
good catch thanks